### PR TITLE
Add game dashboard on home page and project manager

### PR DESCRIPTION
**Pull request by AlexandreSi**
This PR adds a new tab on the homepage to see the games dashboard from there (it displays the games list by default)
It also adds an item in the project manager that displays the game details of the opened project in a dialog.

Also:
- when initial-dialog is used to open the details of a game, use the games dashboard displayed in the homepage
- When right clicking on a project in the list and choosing "Manage game", the games dashboard from the homepage is opened

#### When the opened project has already been published

image

#### When the user is not logged in nor the project has already been published

image

#### When the user is logged in

image

image

#### When the user is not logged in

image

#### Desktop on the homepage menu

image

#### Mobile on the project manager

image

### CODE REVIEW

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/SectionContainer.js [hunk 1]
```diff
@@ -90,17 +90,17 @@ const SectionContainer = ({
...
+            <Line>
+              <TextButton
+                onClick={backAction}
+                icon={<ArrowLeft fontSize="small" />}
+                label={<Trans>Back</Trans>}
+              />
+            </Line>
+          )}
           {title && (
             <SectionRow>
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/SectionContainer.js:94]: No, it seemed okay to me

File: newIDE/app/src/GameDashboard/GameDetails.js [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+  const updateGameFromProject = async (
+    partialGameChange: PartialGameChange,
+    i18n: I18nType
+  ): Promise<boolean> => {
+    if (!project || !profile) return false;
+    const { id } = profile;
+
+    const ownerIds = partialGameChange.ownerIds;
+    if (!ownerIds || !ownerIds.length) {
+      showWarningBox(
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/GameDashboard/GameDetails.js:189]: Good idea, I'll add it

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/ManageSection/index.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+          </Column>
+        )}
+      </SectionRow>
+    </SectionContainer>
+  );
+};
+
+const ManageSectionWithErrorBoundary = (props: Props) => (
+  <ErrorBoundary
+    componentTitle={<Trans>Team section</Trans>}
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/ManageSection/index.js:233]: Thanks for spotting that!

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js [hunk 4]
```diff
@@ -440,6 +440,14 @@ export const HomePage = React.memo<Props>(
         [authenticated]
       );
 
+      const handleGameUpdated = React.useCallback(
+        (game: Game) => {
+          onGameUpdated(game);
+          fetchGames();
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:446]: why is this needed? shouldn't `onGameUpdate()` be in charge of fetching the games again if needed?

Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:446]: I reused the logic that was there before:
One updates the game directly in the state.
The other one fetches the games.
We need to fetch games because there is a logic that sets the chosen game to null in order to show the loader and then the API call comes back and the chosen game is set back to the right one.
I'll look into it tomorrow with fresh eyes, maybe there's more straightforward

File: newIDE/app/src/UI/HighlightingTooltip.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                  {title}
+                </Text>
+                <IconButton size="small" onClick={onClose}>
+                  <Cross fontSize="small" />
+                </IconButton>
+              </Line>
+              {thumbnailSource && (
+                <CorsAwareImage
+                  src={thumbnailSource}
+                  alt="hello"
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/UI/HighlightingTooltip.js:163]: Thanks!

File: newIDE/app/src/Utils/UseDisplayNewFeature.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+type Feature = string;
+
+const useDisplayNewFeature = () => {
+  const {
+    values: { newFeaturesAcknowledgements },
+    setNewFeaturesAcknowledgements,
+  } = React.useContext(PreferencesContext);
+
+  const shouldDisplayNewFeatureHighlighting = React.useCallback(
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/Utils/UseDisplayNewFeature.js:21]: this hook implements the logic to know if the tooltip should be shown. @LuniMoon wants to display it once, then another time 1 week later.
I didn't want to put this logic in the preferences context, I felt like it was not its responsibility

[the remark this answers is not in the document]
Comment (ClementPasteau) [reply newIDE/app/src/Utils/UseDisplayNewFeature.js:21]: It's fine here 👍 

I'm unsure about the value of displaying it twice, especially if we show it on the project manager AND the homepage? so 4 tooltips?

[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/Utils/UseDisplayNewFeature.js:21]: In case you close it without reading it. And it shows at 2 different places it seems fine to me

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ProjectFileListItem.js [hunk 7]
```diff
@@ -311,15 +313,13 @@ export const ProjectFileListItem = ({
...
       actions = actions.concat([
         { type: 'separator' },
         {
           label: i18n._(t`Manage game`),
-          click: () =>
-            navigateToRoute('games-dashboard', {
-              'game-id': gameId,
-            }),
+          click: () => onManageGame({ gameId }),
+          enabled: canManageGame({ gameId }),
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ProjectFileListItem.js:322]: should the whole concat not happen if `!canManageGame( {gameId})` ? It would avoid showing an item that's not clickable?

**Review by ClementPasteau — APPROVED** [review]
nice!!!

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ProjectFileListItem.js (hunk 7 above)
Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ProjectFileListItem.js:322]: I usually prefer disabling items rather than having users think "there's a bug, this item should be here but it's not"
